tensorflow/core/platform/posix/port.cc

```diff
@@ -74,6 +74,14 @@ int NumSchedulableCPUs() {
   return kDefaultCores;
 }
 
+int NumHyperthreadsPerCore() {
```

Comment:
I thin we should implement this properly in tensorflow/core/platform/cpu_info.{h,cc}, i.e. query the relevant bit fields in the cpu info). Here's a stack overflow article with example code: https://stackoverflow.com/questions/2901694/programmatically-detect-number-of-physical-processors-cores-or-if-hyper-threadin/2921632#2921632

Hmmm, wait a second, you're the Intel engineer, you probably know how to do this even better :-)

Reply:
I was going to go down the CPUID route but wanted to check if 'hwloc' (https://www.open-mpi.org/projects/hwloc/ BSD license) was acceptable instead. With that we dont have to maintain the code inside tensorflow. 

If we decide to enumerate cpuid instead, this is the best source so far for Intel CPUs: https://software.intel.com/en-us/articles/intel-64-architecture-processor-topology-enumeration). Need to look into what AMD cpus use.

Reply:
@rmlarsen added code to extract SMT contexts per core on modern intel cpus. can you take a look now?

Reply:
This change appears to have broken my build on Windows.  I get LNK2019 on tensorflow::Port::NumHyperthreadsPerCore. Verified that the commit preceding this builds fine. (2058a241f159a9a6658d48156cca2a2906e1d712).